features/step_definitions/usage_limit_steps.rb

```diff
@@ -51,13 +51,17 @@ def refute_metric_within_usage_limits(metric)
 
 Then "{plan} should have a usage limit of {int} for metric {string} per {string}" do |plan, value, metric_name, period|
   wait_for_requests
-  metric = plan.service.metrics.find_by_system_name!(metric_name)
-  assert_not_nil plan.usage_limits.find_by_metric_id_and_period_and_value(metric.id, period, value)
+  retry_assertions(times: 5) do
+    metric = plan.service.metrics.find_by_system_name!(metric_name)
+    assert_not_nil plan.usage_limits.find_by_metric_id_and_period_and_value(metric.id, period, value)
+  end
```

Comment:
I don't like the idea of retrying assertions. Why do they need to be retried? Ideally, they should just pass, if they fail it's because the test failed. What's the scenario here? why can't the assertion be reliably fixed in some other way?

Reply:
OK fine. 👍